sidechain/contracts/group/GroupConnector.sol

```diff
@@ -0,0 +1,567 @@
+    // REVOKE_INVITATION
+    //////////////////////////////////////////////////
+
+    /**
+     *  [META-TX PREPARE] prepareRevokeInvitation
+     *  TODO
+     *
+     */
+
+    function prepareRevokeInvitation(
+        uint256 _groupId,
+        bytes32 _secretHash,
+        uint256 _nonce
+    )
+        external
+        view
+        returns (bytes32)
+    {
+        // generate unique method call message
+        return keccak256(
+            abi.encodePacked(
+                address(this),
+                "revokeInvitation",
+                _groupId,
+                _secretHash,
+                _nonce
+            )
+        );
+    }
+
+    /**
+     *  [META-TX] revokeInvitation
+     *  TODO
+     *
+     */
+
+    function revokeInvitation(
+        uint256 _groupId,
+        bytes32 _secretHash,
+        bytes calldata _signature,
+        uint256 _nonce
+    )
+        external
+        returns (bool)
+    {
+        address signer = getSigner(
+            this.prepareRevokeInvitation(
+                _groupId,
+                _secretHash,
+                _nonce
+            ),
+            _signature,
+            _nonce
+        );
+
+        // call the logic
+        revokeInvitation(signer, _groupId, _secretHash);
+    }
+
+    /**
+     *  [DIRECT-TX] revokeInvitation
+     *  TODO
+     *
+     */
+
+    function revokeInvitation(
+        uint256 _groupId,
+        bytes32 _secretHash
+    )
+        external
+        returns (bool)
+    {
+        revokeInvitation(msg.sender, _groupId, _secretHash);
+    }
+
+    //////////////////////////////////////////////////
+    // ACCEPT_INVITATION
+    //////////////////////////////////////////////////
+
+    /**
+     *  [META-TX PREPARE] prepareAcceptInvitationCommit
+     *
+     *
+     */
+
+    function prepareAcceptInvitationCommit(
+        uint256 _groupId,
+        bytes32 _addressSecretHash,
+        uint256 _nonce
+    )
+        external
+        view
+        returns (bytes32)
+    {
+        return keccak256(
+            abi.encodePacked(
+                _groupId,
+                "acceptInvitationCommit",
+                _addressSecretHash,
+                _nonce
+            )
+        );
+    }
+
+    /**
+     *  [META-TX] acceptInvitationCommit
+     */
+
+    function acceptInvitationCommit(
+        uint256 _groupId,
+        bytes32 _addressSecretHash,
+        bytes calldata _signature,
+        uint256 _nonce
+    )
+        external
+        returns (bool)
+    {
+        address signer = getSigner(
+            this.prepareAcceptInvitationCommit(
+                _groupId,
+                _addressSecretHash,
+                _nonce
+            ),
+            _signature,
+            _nonce
+        );
+
+        acceptInvitationCommit(signer, _groupId, _addressSecretHash);
+    }
+    
+    /**
+     *  [DIRECT-TX] acceptInvitationCommit
+     */
+
+    function acceptInvitation(
+        uint256 _groupId,
+        bytes32 _secret
```

Comment:
the attribute `_account` is missing. We need it in order to recalculate `sha3(address, secret)` and compare `addressSecretHash`

Reply:
resolved